### PR TITLE
Fixes #24870 - show a setting does not have a default value

### PR DESCRIPTION
**Pull request by boaz0**
I have no idea how to change `Empty` to something else. Could it be that `x-editable-rails` does that?

// cc @ares

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -296,7 +296,17 @@ def self.column_check(opts)
...
-    _("%{full_name} (Default: %{default})") % {full_name: _(full_name), default: default}
+    _("%{full_name} (Default: %{default})") % {full_name: _(full_name), default: has_default? ? default : "Not set" }
+  end
+
+  def has_default?
+    case self.settings_type
+    when "array", "string", "hash"
+      !self.default.empty?
+    else
+      # types integer and boolean will always have a default: 0, false
```
Comment (lzap) [inline app/models/setting.rb]: Nitpick: boolean default value can be either true or false, depending on what you define. The same for integers.

**Comment by lzap** [comment]
Looks good, but would you mind changing the placeholder text just to the default value instead of the title as we do this today? That's weird behavior. This is how you do this:

```diff
diff --git a/app/helpers/application_helper.rb b/app/helpers/application_helper.rb
index 5a081a212..3e0a8ddc7 100644
--- a/app/helpers/application_helper.rb
+++ b/app/helpers/application_helper.rb
@@ -445,9 +445,10 @@ module ApplicationHelper
     update_url = options[:update_url] || url_for(object)
     type       = options[:type]
     title      = options[:title]
+    placeholder = options[:placeholder]
     select_values = [true, false].include?(value) ? [_('Yes'), _('No')] : options[:select_values]
 
-    editable(object, property, {:type => type, :title => title, :value => value, :class => klass, :source => select_values, :url => update_url}.compact)
+    editable(object, property, {:type => type, :title => title, :value => value, :class => klass, :source => select_values, :url => update_url, :placeholder => placeholder}.compact)
   end
 
   def documentation_url(section = "", options = {})
diff --git a/app/helpers/settings_helper.rb b/app/helpers/settings_helper.rb
index f484cacde..2e0134aaf 100644
--- a/app/helpers/settings_helper.rb
+++ b/app/helpers/settings_helper.rb
@@ -10,7 +10,7 @@ module SettingsHelper
         {:title => setting.full_name_with_default, :select_values => self.send("#{setting.name}_collection") })
     end
 
-    return edit_textarea(setting, :value, {:title => setting.full_name_with_default, :helper => :show_value}) if setting.settings_type == 'array'
+    return edit_textarea(setting, :value, {:title => setting.full_name_with_default, :helper => :show_value, :placeholder => "DFFDSD"}) if setting.settings_type == 'array'
     edit_textfield(setting, :value, {:title => setting.full_name_with_default, :helper => :show_value})
   end
```

Other than that, nice work.

**Comment by boaz0** [comment]
@lzap thanks a lot for the review. :heart: 
I updated the code, let me know what you think. :1st_place_medal:

**Comment by boaz0** [comment]
CI failures are unrelated

**Comment by lzap** [comment]
Thanks, I still see for example "HTTP(S) proxy except hosts (Default: [])" both in label and default value while I'd expect only the string `[]` to be in the default value. That's much nicer user experience, this is what I was hoping for if that was not clear. The same for integer.

**Comment by boaz0** [comment]
Hi @lzap 
Can you send me a screen shot, because I am not sure I see this problem.

This is how it looks in my environment:
screenshot-2018-10-2 settings 2

Maybe you want it to look like this?

screenshot-2018-10-2 settings

**Comment by lzap** [comment]
Yes, that's what I'd expect. It looks it works fine for numbers, strings or even arrays:

2018-10-05-115347_1406x323_scrot

However when I try something which has "Not set" I think this is a bit confusing:

2018-10-05-115318_808x317_scrot

I'd expect just to have "Not set" rendered or what you propose "No default value was set". That would be awesome!

2018-10-05-115215_1137x523_scrot

**Comment by boaz0** [comment]
ping @lzap PTAL

**Comment by ohadlevy** [comment]
does it makes sense to combine with #6114 ?

**Comment by boaz0** [comment]
@ohadlevy  I guess we can.

**Comment by lzap** [comment]
What you mean by combining? We already have two issues in redmine etc, I am gonna go ahead and test both, then merge the more simple one and one of the two guys will need to rebase I guess. Or explain please :-)

Edit: I think I am getting it, the other patch to use `has_default?` helper.

**Review by lzap — APPROVED** [review]
That's a go! Thank you.

**Comment by lzap** [comment]
Thank you very much for your contribution, this is now merged.

**Pull request merged**